components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -44,41 +44,45 @@ using mojom::CharacterType;
 using mojom::ConversationTurn;
 
 base::Value::List BuildMessages(
-    const std::string& human_input,
     const std::string& page_content,
     const std::optional<std::string>& selected_text,
     const bool& is_video,
     const EngineConsumer::ConversationHistory& conversation_history) {
-  const std::string prompt_segment_article =
-      page_content.empty()
-          ? ""
-          : base::StrCat(
-                {base::ReplaceStringPlaceholders(
-                     l10n_util::GetStringUTF8(
-                         is_video ? IDS_AI_CHAT_LLAMA2_VIDEO_PROMPT_SEGMENT
-                                  : IDS_AI_CHAT_LLAMA2_ARTICLE_PROMPT_SEGMENT),
-                     {page_content}, nullptr),
-                 "\n\n"});
-
-  std::string date_and_time_string =
-      base::UTF16ToUTF8(TimeFormatFriendlyDateAndTime(base::Time::Now()));
-  std::string system_message = base::StrCat(
-      {prompt_segment_article,
-       base::ReplaceStringPlaceholders(
-           l10n_util::GetStringUTF8(IDS_AI_CHAT_LLAMA2_SYSTEM_MESSAGE_GENERIC),
-           {date_and_time_string}, nullptr)});
-
-  // build
   base::Value::List messages;
 
-  // append system message
+  // Append system message
   {
+    std::string date_and_time_string =
+        base::UTF16ToUTF8(TimeFormatFriendlyDateAndTime(base::Time::Now()));
+    std::string system_message = base::StrCat({base::ReplaceStringPlaceholders(
+        l10n_util::GetStringUTF8(IDS_AI_CHAT_LLAMA2_SYSTEM_MESSAGE_GENERIC),
+        {date_and_time_string}, nullptr)});
+
     base::Value::Dict message;
     message.Set("role", "system");
     message.Set("content", system_message);
     messages.Append(std::move(message));
   }
 
+  // Append page content, if exists
+  if (!page_content.empty()) {
+    const std::string prompt_segment_article = base::ReplaceStringPlaceholders(
+        l10n_util::GetStringUTF8(
+            is_video ? IDS_AI_CHAT_LLAMA2_VIDEO_PROMPT_SEGMENT
+                     : IDS_AI_CHAT_LLAMA2_ARTICLE_PROMPT_SEGMENT),
+        {page_content}, nullptr);
+
+    auto human_turn_iter = base::ranges::find_if(
```

Comment:
Will this find only the first User (Human) message? And prepend the page_content to that message

Reply:
yes